model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -126,24 +145,70 @@ def get_configs(self) -> Generator[RunConfig, None, None]:
         # TODO: TMA-1885: Need an early exit strategy
         for _ in range(n_trials):
             trial = self._study.ask()
-            self._create_trial_objectives(trial)
-            run_config = self._create_objective_based_run_config()
+            trial_objectives = self._create_trial_objectives(trial)
+            run_config = self._create_objective_based_run_config(trial_objectives)
             yield run_config
             score = self._calculate_score()
             self._study.tell(trial, score)
 
-    def _create_trial_objectives(self, trial) -> None:
-        # TODO: TMA-1925: Use SearchParameters here
-        self._instance_count = trial.suggest_int("instance_count", 1, 8)
-        self._batch_size = int(2 ** trial.suggest_int("batch_size", 1, 10))
+    def _create_trial_objectives(self, trial: optuna.Trial) -> TrialObjectives:
+        trial_objectives: TrialObjectives = {}
+        for parameter_name in OptunaRunConfigGenerator.optuna_parameter_list:
+            parameter = self._search_parameters.get_parameter(parameter_name)
+
+            if parameter:
+                trial_objectives = self._add_trial_objective(
```

Comment:
This takes and returns (the same) dict? Is there a cleaner way to do that? I can see us getting burned by this. Maybe just return a single trial objective? And accumulate them in a list/dict?

Reply:
Done